packages/Angular/explorer-core/src/lib/list-component/list-component.ts

```diff
@@ -0,0 +1,200 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { Metadata, RunView } from '@memberjunction/core';
+import { ListEntity } from '@memberjunction/core-entities'; 
+import { NotificationService } from "@progress/kendo-angular-notification";
+
+export class List {
+  ID: number = 0;
+  Name: string = '';
+  Description: string | null = '';
+  Entity: string | null = '';
+  EntityID: number | null = 0;
+  User: string = ''
+}
+
+interface Entity {
+  id: number | null,
+  name: string | null,
+}
+
+@Component({
+  selector: 'mj-list-component',
+  templateUrl: './list-component.html',
+  styleUrls: ['./list-component.css', '../../shared/first-tab-styles.css'],
+})
+
+export class ListComponent implements OnInit {
+  private md = new Metadata();
+  private entities: Array<Entity> = [];
+  private list: Array<List> = [];
+  private selectedEntity = new List();
+  private selectedRow: List | null = null;
+  protected _contextUser: object | null;
+  public opened: boolean = false;
+  public deleteDialogOpened: boolean = false;
+  public selectedEntities: Array<Entity> = [];
+  public mySelection: string[] = [];
+  public loader: boolean = false;
+  public btnLoader: boolean = false;
+  public listName: string = ''
+  public listDescription: string | null = ''
+  public title: string = ''
+
+  constructor(private notificationService: NotificationService, private router: Router) {
+    this._contextUser = null
+   }
+
+  get getEntities() {
+    return this.entities
+  }
+
+  get getList() {
+    return this.list
+  }
+
+  private appendObj(listObj: List) {
+    const transformed = {
+      ID: listObj.ID,
+      Name: listObj.Name,
+      Description: listObj.Description,
+      Entity: listObj.Entity,
+      EntityID: listObj.EntityID,
+      User: listObj.User
+    }
+    if(this.selectedRow) {
+      const index: number = this.list.findIndex((item) => item.ID === listObj.ID)
+      Object.assign(this.list[index], transformed)
+    }
+    else this.list = [ ...this.list, transformed]
+       
+  }
+
+  private updateList(listID: number) {
+    this.list = this.list.filter((item) => item.ID !== listID)
+  }
+
+  private resetForm() {
+    this.listName = ''
+    this.listDescription = ''
+    this.selectedEntities = []
+    this.selectedRow = null
+  }
+
+  async ngOnInit(): Promise<void> {
+    await this.loadLists()
+    await this.loadEntities()
+  }
+
+  public async loadLists(): Promise<void> {
+    this.loader = true
+    const rv = new RunView();
+    const models = await rv.RunView({EntityName: 'Lists', ResultType: 'entity_object'}, this.md.CurrentUser);
+    this.list = models?.Results.map((entity) => {
+    return {
```

Comment:
Because the result type has the value `entity_object`, then the values returned will be of the type `ListEntity` meaning you dont need to map the data to anything else and use the records directly.
i.e. do `const listEntities: ListEntity = models.Results` instead